SwiftUIDZ/SwiftUIDZ/Helpers/LockerSlider.swift

```diff
@@ -0,0 +1,186 @@
+                                )
+                        }.frame(width: 7, height: 25)
+                        ZStack {
+                            RoundedRectangle(cornerRadius: 3)
+                                .fill(Color("buttonBackground").opacity(0.5))
+                                .overlay(
+                                    RoundedRectangle(cornerRadius: 2)
+                                        .stroke(Color.black.opacity(1), lineWidth: 1)
+                                        .blur(radius: 1)
+                                        .offset(x: 2, y: 2)
+                                        .mask(RoundedRectangle(cornerRadius: 3).fill(LinearGradient(Color.black, Color.clear)))
+                                )
+                                .overlay(
+                                    RoundedRectangle(cornerRadius: 3)
+                                        .stroke(Color.white.opacity(0.15), lineWidth: 1)
+                                        .blur(radius: 1)
+                                        .offset(x: -2, y: -2)
+                                        .mask(RoundedRectangle(cornerRadius: 3).fill(LinearGradient(Color.clear, Color.black)))
+                                )
+                        }.frame(width: 7, height: 25)
+                    }
+                }
+                
+                
+                
+                
+                
+                
+                    .frame(width: length, height: length/1.7)
+                    .offset(x: (proxy.size.width - length) * ratio)
+                    .shadow(color: Color.black.opacity(0.5), radius: 5, x: 2, y: 2)
+                    .gesture(DragGesture(minimumDistance: 0)
+                                .onChanged({ updateStatus(value: $0, proxy: proxy) })
+                                .onEnded { _ in startX = nil })
+            }
+            .frame(height: length)
+            .simultaneousGesture(DragGesture(minimumDistance: 0)
+                                    .onChanged({ update(value: $0, proxy: proxy) }))
+            .onAppear {
+                ratio = min(1, max(0,CGFloat(value / bounds.upperBound)))
+            }
+        }
+    }
+
+    // MARK: Private
+    private var overlay: some View {
+        RoundedRectangle(cornerRadius: (length + lineWidth) / 2)
+            .stroke(Color.gray, lineWidth: lineWidth)
+            .frame(height: length + lineWidth)
+    }
+
+
+    // MARK: - Function
+    // MARK: Private
+    private func updateStatus(value: DragGesture.Value, proxy: GeometryProxy) {
+        guard startX == nil else { return }
+    
+        let delta = value.startLocation.x - (proxy.size.width - length) * ratio
+        startX = (length < value.startLocation.x && 0 < delta) ? delta : value.startLocation.x
+    }
+
+    private func update(value: DragGesture.Value, proxy: GeometryProxy) {
```

Comment:
```suggestion
    private func update(value: DragGesture.Value, proxy: GeometryProxy) {
```
логику во вью модель